vcpkg/concepts/registries.md

```diff
@@ -3,49 +3,404 @@ title: Registries concepts
 description: Concepts about vcpkg registries and their capabilities.
 author: vicroms
 ms.author: viromer
-ms.date: 01/10/2024
+ms.date: 11/10/2024
 ms.topic: concept-article
 ---
-
 # Registries concepts
 
-Registries are collections of ports and their versions. The curated registry is
-the one hosted at <https://github.com/Microsoft/vcpkg>. vcpkg lets you create
-custom registries, which may be hosted by a variety of public or private providers.
+## Overview
+
+vcpkg hosts a selection of libraries packaged into [ports](ports.md) at
+<https://github.com/Microsoft/vcpkg>, this collection of ports is called the
+*curated registry*. However, vcpkg is not limited to just installing ports from
+the curated registry. Users are capable of extending the selection of ports by
+creating custom registries.
+
+A registry is a collection of ports and helper files arranged in a specific
+structure. By following the registry structure, ports contained in a registry
+can be installed with the same amount of features
+([versioning](../users/versioning.md), [binary
+caching](../users/binarycaching.md)) offered to ports in the curated registry.
+
+There are currently three kinds of registries: 
+
+* The [built-in registry](#built-in-registry),
+* [Git registries](#git-registries), and 
+* [filesystem registries](#filesystem-registries).
+
+vcpkg can be instructed to consider ports available in custom registries by
+using a [`vcpkg-configuration.json`
+file](../reference/vcpkg-configuration-json.md#registries). See the [Tutorial:
+Install a dependency from a Git-based registry](../consume/git-registries.md)
+article for a tutorial on how to use custom registries in your projects.
+
+## Registry structure
+
+For vcpkg to interface with a registry, it must comform with the structure
+below:
+
+* A directory named `ports`, contains the collection of ports, with each
+  subdirectory containing a specific port matching the subdirectory name. For
+  example, the files for port `foo` must be located in `ports/foo`.
+* A directory named `versions` must contain the files that comprise the [versions
+  database](#versions-database).
+  
+### Example: registry structure
+
+```
+ports/
+  foo/
+    portfile.cmake
+    vcpkg.json
+versions/
+  f-/
+    foo.json
+  baseline.json
+```
+
+## Versions database
+
+All registries, regardless of their kind, must contain a `versions` folder at the
+root of the registry which contains the _versions database_.
+
+There are two components to the versions database:
+
+* the baseline file, 
+* the version files.
+
+The baseline file is a JSON file named `baseline.json` located at the root of
+the `versions` directory. The purpose of the baseline file is to describe the
+set of versions that are considered to be the latest for each port in the
+registry. The layout of the baseline file depends on the kind of registry.
```

Comment:
```suggestion
the `versions` directory. The baseline file describes the
set of versions that are current for each port in the
registry. The layout of the baseline file depends on the kind of registry.
```

I suggest 'current' rather than 'latest' because there is no requirement that the versions listed in the baseline are the ones that sort last.